codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -1113,11 +1117,11 @@ let rollback = (self: t, ~lastScannedBlock, ~firstChangeEvent) => {
 * Returns a boolean indicating whether the fetch state is actively indexing
 * used for comparing event queues in the chain manager
 */
-let isActivelyIndexing = ({baseRegister}: t) => {
+let isActivelyIndexing = ({baseRegister} as fetchState: t, ~endBlock) => {
   // nesting to limit additional unnecessary computation
-  switch baseRegister.registerType {
-  | RootRegister({endBlock: Some(endBlock)}) =>
-    let isPastEndblock = baseRegister.latestFetchedBlock.blockNumber >= endBlock
+  switch (baseRegister.registerType, endBlock) {
+  | (RootRegister, Some(endBlock)) =>
+    let isPastEndblock = getLatestFullyFetchedBlock(fetchState).blockNumber >= endBlock
```

Comment:
This `getLatestFullyFetchedBlock` usage is the only change related to the fix. Since we want to consider pending dynamic contracts as well.

Other changes are either renaming of `fetchState` to `partitionedFetchState` which is more correct, since these are two different entities. And removing endBlock from `fetchState` since it's not needed there.

Reply:
I'm not 100% following the intricacies to be honest. Are you saying that before this fix, there could be pending requests to fetch more events for dynamic contracts that weren't accounted for - and 'getLatestFullyFetchedBlock' will include that?

Reply:
Exactly